contracts/modules/Affiliates.sol

```diff
@@ -0,0 +1,278 @@
+		result.alreadySet = affiliatesUserReferrer[user] != address(0) ? true : false;
+		result.success = !(result.userNotFirstTradeFlag || result.alreadySet || user == referrer);
+		if (result.success) {
+			affiliatesUserReferrer[user] = referrer;
+			referralsList[referrer].add(user);
+			emit SetAffiliatesReferrer(user, referrer);
+		} else {
+			emit SetAffiliatesReferrerFail(user, referrer, result.alreadySet, result.userNotFirstTradeFlag);
+		}
+	}
+
+	function getReferralsList(address referrer) public view returns (address[] memory refList) {
+		refList = referralsList[referrer].enumerate();
+		return refList;
+	}
+
+	function getUserNotFirstTradeFlag(address user) public view returns (bool) {
+		return userNotFirstTradeFlag[user];
+	}
+
+	//REFACTOR move setUserNotFirstTradeFlag to ProtocolSettings?
+	function setUserNotFirstTradeFlag(address user) external onlyCallableByLoanPools {
+		if (!userNotFirstTradeFlag[user]) {
+			userNotFirstTradeFlag[user] = true;
+			emit SetUserNotFirstTradeFlag(user);
+		}
+	}
+
+	function _getAffiliatesTradingFeePercentForSOV() internal view returns (uint256) {
+		return affiliateFeePercent;
+	}
+
+	// calculate affiliates trading token fee amount
+	function _getReferrerTradingFeeForToken(uint256 feeTokenAmount) internal view returns (uint256) {
+		return feeTokenAmount.mul(getAffiliateTradingTokenFeePercent()).div(10**20);
+	}
+
+	function getAffiliateTradingTokenFeePercent() public view returns (uint256) {
+		return affiliateTradingTokenFeePercent;
+	}
+
+	function getMinReferralsToPayout() public view returns (uint256) {
+		return minReferralsToPayout;
+	}
+
+	/**
+	 * @dev get the sovToken reward of a trade. The reward is worth x% of the trading fee.
+	 * @param feeToken the address of the token in which the trading/borrowig fee was paid
+	 * @param feeAmount the height of the fee
+	 * */
+	function _getSovBonusAmount(address feeToken, uint256 feeAmount) internal view returns (uint256) {
+		uint256 rewardAmount;
+		address _priceFeeds = priceFeeds;
+
+		//calculate the reward amount, querying the price feed
+		(bool success, bytes memory data) =
+			_priceFeeds.staticcall(
+				abi.encodeWithSelector(
+					IPriceFeeds(_priceFeeds).queryReturn.selector,
+					feeToken,
+					sovTokenAddress, // dest token = SOV
+					feeAmount.mul(_getAffiliatesTradingFeePercentForSOV()).div(10**20)
+				)
+			);
+		assembly {
+			if eq(success, 1) {
+				rewardAmount := mload(add(data, 32))
+			}
+		}
+
+		return rewardAmount;
+	}
+
+	/**
+	 * @param tradingFeeTokenBaseAmount total trading fee amount, the base for calculating referrer's fees
+	 */
+	function payTradingFeeToAffiliatesReferrer(
+		address referrer,
+		address token,
+		uint256 tradingFeeTokenBaseAmount
+	) external onlyCallableInternal returns (uint256 referrerBonusSovAmount, uint256 referrerBonusTokenAmount) {
+		bool isHeld = referralsList[referrer].length() >= getMinReferralsToPayout() ? false : true;
```

Comment:
`bool isHeld = referralsList[referrer].length() < getMinReferralsToPayout();`
in general, `true` and `false` are only required in single-word statements like `return true;`, `while(true)` or `someFlag = false;`